### PR TITLE
made fields final

### LINKED ISSUE #2437: making CompositeDisposable and RealmDataRepository final inside VM
**Expected Behaviour**

These should be assigned only once ie. should be made final inside VM

### PR DESCRIPTION
**Pull request by simarsingh24**
Fixes #2437

### CODE REVIEW

File: android/app/src/main/java/org/fossasia/openevent/core/discount/DiscountFragmentViewModel.java [hunk 1]
```diff
@@ -20,9 +20,9 @@
 public class DiscountFragmentViewModel extends ViewModel {
 
     private LiveData<List<DiscountCode>> discountCodes;
-    private RealmDataRepository realmRepo;
+    private final RealmDataRepository realmRepo;
     private MutableLiveData<Boolean> discountCodesDownloadResponse;
-    private CompositeDisposable compositeDisposable;
+    private final CompositeDisposable compositeDisposable;
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/core/discount/DiscountFragmentViewModel.java]: Pull them together on top

**Review by iamareebjamal — CHANGES_REQUESTED** [review]
Put final fields on top and add a line feed between them

File: android/app/src/main/java/org/fossasia/openevent/core/discount/DiscountFragmentViewModel.java (hunk 1 above)
Comment (simarsingh24) [reply android/app/src/main/java/org/fossasia/openevent/core/discount/DiscountFragmentViewModel.java]: okay

File: android/app/src/main/java/org/fossasia/openevent/core/sponsor/SponsorsFragmentViewModel.java [hunk 2]
```diff
@@ -12,9 +12,9 @@
 
 public class SponsorsFragmentViewModel extends ViewModel {
 
+    private final RealmDataRepository realmRepo;
     private LiveData<List<Sponsor>> sponsorsList;
-    private RealmDataRepository realmRepo;
-
+
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/core/sponsor/SponsorsFragmentViewModel.java]: extra

**Review by iamareebjamal — CHANGES_REQUESTED** [review]
Separate final and non-final fields by a line feed

**Review by nikit19 — APPROVED** [review]

**Review by iamareebjamal — CHANGES_REQUESTED** [review]
Please check previous request for changes

**Comment by iamareebjamal** [comment]
@simarsingh24 Please resolve the conflicts

**Comment by simarsingh24** [comment]
resolve conflicts

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**